lib/presentation/create_plan/create_plan_notifier.dart

```diff
@@ -143,6 +143,27 @@ class CreatePlanNotifier extends _$CreatePlanNotifier {
       );
       return;
     }
+
+    if (state.requireValue.startDate != null &&
+        state.requireValue.endDate != null) {
+      try {
+        final startDate = _parseDate(state.requireValue.startDate!);
+        final endDate = _parseDate(state.requireValue.endDate!);
+
+        if (startDate.isAfter(endDate)) {
+          scaffoldMessenger.showExceptionSnackBar(
+            t.createPlanPage.snackBar.error.invalidDateRange,
+          );
+          return;
+        }
+      } on Exception {
+        scaffoldMessenger.showExceptionSnackBar(
+          t.createPlanPage.snackBar.error.invalidDateRange,
+        );
+        return;
+      }
+    }
```

Comment:
この処理の真上で、全ての選択ができているか確認しているコードがあるね
なのでここで開始日と終了日がnullかの条件はいらないかも

また、try catchは非同期処理で使うことがメインなので、基本はいらなさそう
なので下のような処理になるはず

```suggestion
        final startDate = _parseDate(state.requireValue.startDate!);
        final endDate = _parseDate(state.requireValue.endDate!);

        if (startDate.isAfter(endDate)) {
          scaffoldMessenger.showExceptionSnackBar(
            t.createPlanPage.snackBar.error.invalidDateRange,
          );
          return;
        }
```